src/Signer.js

```diff
@@ -0,0 +1,72 @@
+import { PublishRequest } from 'streamr-client-protocol'
+
+const Web3 = require('web3')
+
+const web3 = new Web3()
+
+const SIGNATURE_TYPE_ETH = 1
+
+export default class Signer {
+    constructor(options) {
+        this.options = options || {}
+        if (this.options.privateKey) {
+            const account = web3.eth.accounts.privateKeyToAccount(this.options.privateKey)
+            this.address = account.address
+            this.sign = (d) => account.sign(d).signature
+        } else if (this.options.provider) {
+            const w3 = new Web3(this.options.provider)
+            const accounts = w3.eth.getAccounts()
+            const address = accounts[0]
+            if (!address) {
+                throw new Error('Cannot access account from provider')
+            }
+            this.address = address
+            this.sign = async (d) => w3.eth.personal.sign(d, this.address)
+        } else {
+            throw new Error('Need either "privateKey" or "provider".')
+        }
+    }
+
+    async signData(data, signatureType = SIGNATURE_TYPE_ETH) {
+        if (signatureType === SIGNATURE_TYPE_ETH) {
+            return this.sign(data)
+        }
+        throw new Error(`Unrecognized signature type: ${signatureType}`)
+    }
+
+    async getSignedPublishRequest(publishRequest, signatureType = SIGNATURE_TYPE_ETH) {
+        const ts = publishRequest.getTimestampAsNumber()
+        if (!ts) {
+            throw new Error('Timestamp is required as part of the data to sign.')
+        }
+        const payload = this.address.toLowerCase() + publishRequest.streamId + ts + publishRequest.getSerializedContent()
```

Comment:
Yes I think a new `signatureType` in case of a new payload to sign makes sense. I would maintain the `signData` because the signing algorithm itself might change and not the payload.